internal/blockchain/ethereum/quorum/geth_provider.go

```diff
@@ -0,0 +1,409 @@
+		Type: "ethereum",
+		Ethereum: &types.EthereumConfig{
+			Ethconnect: &types.EthconnectConfig{
+				URL:   connectorURL,
+				Topic: m.ID,
+			},
+		},
+	}
+	return
+}
+
+func (p *GethProvider) GetOrgConfig(stack *types.Stack, m *types.Organization) (orgConfig *types.OrgConfig) {
+	account := m.Account.(*ethereum.Account)
+	orgConfig = &types.OrgConfig{
+		Name: m.OrgName,
+		Key:  account.Address,
+	}
+	return
+}
+
+func (p *GethProvider) Reset() error {
+	return nil
+}
+
+func (p *GethProvider) GetContracts(filename string, extraArgs []string) ([]string, error) {
+	contracts, err := ethereum.ReadContractJSON(filename)
+	if err != nil {
+		return []string{}, err
+	}
+	contractNames := make([]string, len(contracts.Contracts))
+	i := 0
+	for contractName := range contracts.Contracts {
+		contractNames[i] = contractName
+		i++
+	}
+	return contractNames, err
+}
+
+func (p *GethProvider) DeployContract(filename, contractName, instanceName string, member *types.Organization, extraArgs []string) (*types.ContractDeploymentResult, error) {
+	contracts, err := ethereum.ReadContractJSON(filename)
+	if err != nil {
+		return nil, err
+	}
+	return p.connector.DeployContract(contracts.Contracts[contractName], instanceName, member, extraArgs)
+}
+
+func (p *GethProvider) CreateAccount(args []string) (interface{}, error) {
+	l := log.LoggerFromContext(p.ctx)
+	memberIndex := args[2]
+	memberCount := args[3]
+	gethVolumeName := fmt.Sprintf("%s_geth_%s", p.stack.Name, memberIndex)
+	var directory string
+	stackHasRunBefore, err := p.stack.HasRunBefore()
+	if err != nil {
+		return nil, err
+	}
+	if stackHasRunBefore {
+		directory = p.stack.RuntimeDir
+	} else {
+		directory = p.stack.InitDir
+	}
+
+	prefix := strconv.FormatInt(time.Now().UnixNano(), 10)
+	outputDirectory := filepath.Join(directory, "blockchain", fmt.Sprintf("geth_%s", memberIndex), "keystore")
+	keyPair, walletFilePath, err := ethereum.CreateWalletFile(outputDirectory, prefix, keyPassword)
+	if err != nil {
+		return nil, err
+	}
+
+	// Tessera is an optional add-on to the quorum blockchain node provider
+	var tesseraPubKey, tesseraKeysPath string
+	if p.stack.TesseraEnabled {
+		tesseraVolumeName := fmt.Sprintf("%s_tessera_%s", p.stack.Name, memberIndex)
+		tesseraKeysOutputDirectory := filepath.Join(directory, "tessera", fmt.Sprintf("tessera_%s", memberIndex), "keystore")
+		_, tesseraPubKey, tesseraKeysPath, err = CreateTesseraKeys(p.ctx, tesseraImage, tesseraKeysOutputDirectory, "", "tm", keyPassword)
+		if err != nil {
+			return nil, err
+		}
+		l.Info(fmt.Sprintf("keys generated in %s", tesseraKeysPath))
+		l.Info(fmt.Sprintf("generating tessera docker-entrypoint file for member %s", memberIndex))
+		tesseraEntrypointOutputDirectory := filepath.Join(directory, "tessera", fmt.Sprintf("tessera_%s", memberIndex))
+		if err := CreateTesseraEntrypoint(p.ctx, tesseraEntrypointOutputDirectory, tesseraVolumeName, memberCount, p.stack.Name); err != nil {
```

Comment:
I get that the TesseraEntrypoint is quite tied to the creation of the account with tessera keys but the CLI has followed a pattern where the blockchain configuration is stored in the `stack` directory such as the genesis and then at `ff start` it gets copied over so we don't envision anyone modifying this Tessera file and then restarting the stack? It won't get copied over again based on where this code is

Reply:
Refactored for docker entry point files to follow the same CLI pattern of storing them in the `stack` directory. I got pretty confused over here, are you saying if we follow the CLI pattern, we are able to modify the docker-entrypoint file and use the new file whenever we stop and start the stack?